File: src/main/java/org/example/inventorysystem/models/Type.java
package org.example.inventorysystem.models;

import lombok.Getter;

@Getter
public enum Type {
	Computer("Computer asset", 487),
	Furniture("Furniture asset", 809),
	Measure("Measure asset", 664),
	Audiovisual("Audiovisual asset", 662),
	Intelectual("Intelectual asset", 0202);


	private final String description;
	private final int number;
	Type(String description, int number){
		this.description = description;
		this.number = number;
	}

	public static Type getGroup(int number){
		for(Type type : Type.values()){
			if(type.getNumber() == number){
				return type;
			}
		}
		throw new IllegalArgumentException("No group with number " + number + " found");
	}

	public static String getDescription(int number){
		for(Type type : Type.values()){
			if(type.getNumber() == number){
				return type.getDescription();
			}
		}
		throw new IllegalArgumentException("No group with number " + number + " found");
	}
}

File: inventory_front/src/app/type.ts
export enum TypeEnum {
  Computer = 487,
  Furniture = 809,
  Measure = 664,
  Audiovisual = 662,
  Intelectual = 202
}

export const TypeDescriptions: { [key in TypeEnum]: string } = {
  [TypeEnum.Computer]: "Computer asset",
  [TypeEnum.Furniture]: "Furniture asset",
  [TypeEnum.Measure]: "Measure asset",
  [TypeEnum.Audiovisual]: "Audiovisual asset",
  [TypeEnum.Intelectual]: "Intelectual asset",
};

export function getDescription(type: TypeEnum): string {
  return TypeDescriptions[type];
}
